[Ka Shing] iP

kashing555 opened the pull request:
# **DUKE**

Duke frees your mind of having to remember things you need to do. It's:

1. text-based
2. easy to learn
3. Very fast to use

All you need to do is:

1. download it from [here](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!
Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

Commit:
Add Gradle support

kashing555 committed:
Level-1

kashing555 committed:
Level-2

kashing555 committed:
Level-3

kashing555 committed:
Level-4

kashing555 committed:
A-TextUiTesting

kashing555 committed:
Level-5

kashing555 committed:
Level-6

kashing555 committed:
Updated If else statements in Duke.Main to switch cases and additional Exceptions.

kashing555 committed:
Updated DukeException to switch cases

kashing555 committed:
branch-Level-7

kashing555 committed:
branch-Level-8

kashing555 committed:
Merge tag 'branch-Level-8'

kashing555 committed:
A-MoreOOP

kashing555 committed:
A-Packages

kashing555 committed:
A-JUnit

kashing555 committed:
A-Jar

kashing555 committed:
A-JavaDoc

kashing555 committed:
A-CodingStandard

kashing555 committed:
Level-9

kashing555 committed:
Merge branch 'branch-A-CodingStandard'

kashing555 committed:
Merge branch 'branch-Level-9'

kashing555 committed:
Added additional header comments, fixed method and parameter having the same name.

putaojuice commented on src/main/java/commands/CreateTask.java:
```diff
@@ -0,0 +1,42 @@
+     * @param list contains the list of tasks that are currently being tracked.
+     */
+    public static void deadline(String description, List<Task> list) {
+        String[] restOfPara  = description.split("/by ", 2);
```
Not sure if I like the variable name because Para can be inferred as paragraph, parameter and etc. Might be better if you just spell out the whole word for readability.

putaojuice commented on src/main/java/commands/CreateTask.java:
```diff
@@ -0,0 +1,42 @@
+     * @param list contains the list of tasks that are currently being tracked.
+     */
+    public static void event(String description, List<Task> list) {
+        String[] restOfPara2 = description.split("/at ", 2);
```
Why do you have to name `restOfPara2` with a 2?

putaojuice commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,119 @@
+    List<String> inputList = new ArrayList<>(Arrays.asList(validInputs));
+    String[] validInputs2 = new String[] {"mark", "unmark", "delete"};
+    List<String> inputList2 = new ArrayList<>(Arrays.asList(validInputs2));
+    String[] validInputs3 = new String[] {"list", "bye"};
```
I am not sure if it is needed for you to split all the valid inputs into 3 different `List`.

putaojuice commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    /**
+     * The savTasks method saves the list of tasks that are currently being tracked into a txt file.
```
Small spelling mistake in your header.

putaojuice commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,98 @@
+
+        System.out.println("------------------------------------------------------------");
+        switch (tempList[0]) {
+            case "bye":
```
Shouldn't there be no indentation for case clauses?

putaojuice commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,98 @@
+     * Shows the number of task being track.
+     */
+    private static void printAndSave(int taskCount) {
+        if (taskCount + 1 == 1) {
```
Not sure if you need to keep track of the `taskCount` because Java `List` should already do the job for you?

putaojuice commented on src/main/java/myTasks/TaskList.java:
```diff
@@ -0,0 +1,35 @@
+                case "D":
+                    holder = new Deadline(tempList[2], tempList[3]);
+                    break;
+                default:
```
Shouldn't you add another case where it is `"E"`?

putaojuice reviewed (commented):
Good job 👍 ! Looks good in general, the structure of your code is clear. No heavily nested logic clauses spotted. 

Just some things to note. Spotted some Java coding style violation on switch-case usage and some cases where you name your variables with 1, 2, 3 at the back which could cause confusion.

zxgoh commented on src/main/java/commands/CreateTask.java:
```diff
@@ -0,0 +1,42 @@
+     * @param description contains the description of the task.
+     * @param list contains the list of tasks that are currently being tracked.
+     */
+    public static void todo(String description, List<Task> list) {
```
Perhaps a more meaningful name could be used instead of `list`, such as:
- `todoList`
- `taskList`
- etc

zxgoh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,83 @@
+     */
+    public void saveTasks(List<Task> list) {
+        try {
+            FileWriter myWriter = new FileWriter(this.filepath);
```
Great use of FileWriter 👍

zxgoh commented on src/main/java/myTasks/Event.java:
```diff
@@ -0,0 +1,23 @@
+package myTasks;
+
+import java.time.*;
```
Perhaps avoid using import wildcards if necessary :grimacing:

zxgoh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,98 @@
+import myTasks.TaskList;
+import myTasks.Task;
+
+import commands.*;
```
Similarly here, avoid use of import wildcards

kashing555 committed:
Merge pull request #2 from kashing555/add-gradle-support

Add Gradle support

kashing555 committed:
A-Gradle

kashing555 committed:
A-CheckStyle

kashing555 committed:
Level-10

kashing555 committed:
Most critical points in the program has been covered by DukeException.

Assert is applied mainly in the MainWindow, Duke and DeleteTask.

kashing555 committed:
Majority of code quality has been done in CheckStyle and Gradle commit but will be mentioned here as well.

Removed redundent parameters being used by classes.
Renamed methods to better represent their usage.
Reduce and abstracted long methods. (SLAP)
Prevent deep nesting by abstracting methods. (SLAP)
Changed comment of methods to write to the reader.

kashing555 committed:
A-BetterGui

kashing555 committed:
Merge pull request #3 from kashing555/A-BetterGui

A-BetterGui

kashing555 committed:
Adding PNG for readme

kashing555 committed:
A-UserGuide

A-UserGuide

kashing555 committed:
Update README.md

kashing555 committed:
Update README.md

kashing555 committed:
Merge pull request #4 from kashing555/branch-A-UserGuide

Branch a user guide

kashing555 committed:
Set theme jekyll-theme-cayman

kashing555 committed:
Update README.md

kashing555 committed:
Minor changes to ReadMe

kashing555 committed:
Merge branch 'master' of https://github.com/kashing555/ip

kashing555 committed:
Update README.md

kashing555 committed:
Showcase not accepting PNG files

Showcase only accept png not PNG

kashing555 committed:
Delete Ui.PNG

kashing555 committed:
Set theme jekyll-theme-slate

kashing555 committed:
Update date time format

kashing555 committed:
Merge branch 'master' of https://github.com/kashing555/ip
